### PR TITLE
Feature/new extensions

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# Reality Collective - Reality Toolkit Pull Request

## Overview

Added / updated the extensions in the Utilities package

## Changes

* Assembly extension - gets loadable types
* Handedness extensions - to standardise some handedness queries
* Added Sprite Extensions to convert a sprite in to a Texture2D
* Added GameObject extension to validate its use

## Breaking Changes

None

### CODE REVIEW

File: Runtime/Extensions/AssemblyExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (FejZa) [inline Runtime/Extensions/AssemblyExtensions.cs:1]: Is this from MRTK?

File: Runtime/Extensions/GameObjectExtensions.cs [hunk 2]
```diff
@@ -236,5 +238,18 @@ public static bool IsParentOrChildOf(this GameObject g1, GameObject g2)
             T component = gameObject.GetComponent<T>();
             return component.IsNull() ? gameObject.AddComponent<T>() : component;
         }
+
+        /// <summary>
+        /// Validates the <see cref="GameObject"/> reference.
+        /// </summary>
+        /// <param name="gameobject">The target <see cref="GameObject"/>.</param>
+        /// <param name="callerName">The <see cref="CallerFilePathAttribute"/> fills in this information.</param>
+        public static void Validate(this GameObject gameobject, [CallerFilePath] string callerName = "", string fieldName = "")
```
Comment (FejZa) [inline Runtime/Extensions/GameObjectExtensions.cs:247]: What in the world does this do?

**Review by FejZa — APPROVED** [review]

File: Runtime/Extensions/GameObjectExtensions.cs (hunk 2 above)
Comment (SimonDarksideJ) [reply Runtime/Extensions/GameObjectExtensions.cs:247]: Allows you to validate if a GameObject reference has been applied in a script and then warns you if something hasn't been configured and on which object.  so in OnValidate you can do something like this:

```csharp
 myGameObjectReference.Validate();
```

This effectively ensures a gameobject has been assigned for a script.

File: Runtime/Extensions/AssemblyExtensions.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply Runtime/Extensions/AssemblyExtensions.cs:1]: Yes, this specific script is.

**Pull request merged**